google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -323,14 +327,12 @@ def _create_and_import(
             credentials=credentials,
         )
 
-        if gcs_source and not is_tabular_dataset_metadata:
-            return dataset_obj.import_data(
-                gcs_source=gcs_source,
-                import_schema_uri=import_schema_uri,
-                data_items_labels=data_items_labels,
-            )
-        else:
-            return dataset_obj
+        # Import if import datasource is NonTabularDatasourceImportable
+        if isinstance(datasource, datasources.NonTabularDatasourceImportable):
```

Comment:
This should depend on the abstraction and not the concrete type (see dependency-inversion principle of SOLID).

So it should just check against `_DatasourceImportable` which is the abstraction that represents importing.